ompi/runtime/ompi_spc.c

```diff
@@ -188,19 +247,95 @@ static int ompi_spc_notify(mca_base_pvar_t *pvar, mca_base_pvar_event_t event, v
     index = (int)(uintptr_t)pvar->ctx;  /* Convert from MPI_T pvar index to SPC index */
 
     /* For this event, we need to set count to the number of long long type
-     * values for this counter.  All SPC counters are one long long, so we
-     * always set count to 1.
+     * values for this counter.  Most SPC counters are one long long so the
+     * default is 1, however bin counters and the xml string can be longer.
      */
-    if(MCA_BASE_PVAR_HANDLE_BIND == event) {
-        *count = 1;
+    do {
+        if(MCA_BASE_PVAR_HANDLE_BIND == event) {
+            /* Convert from MPI_T pvar index to SPC index */
+            index = pvar->pvar_index - mpi_t_offset;
+            if(index < 0) {
+                char *shm_dir;
+                if(0 == access("/dev/shm", W_OK)) {
```

Comment:
Can we use opal_shmem_mmap_backing_file_base_dir instead of the hardcoded /dev/shm ?